src/ape_ethereum/transactions.py

```diff
@@ -207,6 +214,34 @@ def failed(self) -> bool:
     def call_tree(self) -> Optional[CallTreeNode]:
         return self.provider.get_call_tree(self.txn_hash)
 
+    @cached_property
+    def debug_logs(self) -> List[List[Any]]:
+        """
+        Extract messages to console outputted by contracts via print() or console.log() statements
+        """
+        lines: List[List[Any]] = []
+
+        # Some providers do not implement this, so skip
+        try:
+            self.call_tree
+        except APINotImplementedError:
+            logger.debug("Call tree not available, skipping print log extraction")
+            return lines
+
+        # Extract any Vyper print() calls to output
+        for call in extract_prints(self):
+            if call.inputs is not None:
+                lines.append(vyper_print(call.inputs))
+
+        # Extract any Hardhat console.log() calls to output
```

Comment:
thought: The whole reason we made the plugin system is to handle all these one-offs in their respective plugins.
I understand the architecture might not be accommodating for this right now, but it's something to think about. 

Because of suddenly Foundry invents its own log format, now we have to adjust here.

And users who are not using Hardhat, this is still checked for.

Reply:
some complicating factors here: foundry uses hardhat style console.log, and vyper has a compat mode that also exposes it. so, may not be as simple or relegated to the hardhat plugin only, the behavior is more widely adopted across ethereum ecosystem tooling

Reply:
Yeah, it's not really a hardhat thing.  It's just a contract they wrote and others sort of standardized to.  It's very possible to never use hardhat but use their console.sol contract for this stuff.

Reply:
Ah got it, I understand now